src/prommis/cmi_precipitator/tests/test_precipitator.py

```diff
@@ -0,0 +1,226 @@
+#####################################################################################################
+# “PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
+# (“PrOMMiS”) initiative, and is copyright (c) 2023-2025 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory, et al. All rights reserved.
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license information.
+#####################################################################################################
+from pyomo.environ import ConcreteModel, assert_optimal_termination, value
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_total_constraints,
+    number_unused_variables,
+    number_variables,
+)
+from idaes.core.util.scaling import set_scaling_factor
+
+import pytest
+
+from prommis.cmi_precipitator import aqueous_properties as aqueous_thermo_prop_pack
+from prommis.cmi_precipitator import (
+    precipitate_properties as precipitate_thermo_prop_pack,
+)
+from prommis.cmi_precipitator.opt_based_precipitator import Precipitator
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+# -----------------------------------------------------------------------------
+class TestPrec(object):
```

Comment:
I would just add two more small checks to this file: assert_no_numerical_warnings and assert_no_strucutural_warnings. There are plenty of examples for how to write these tests throughout the repo